src/components/main-panel/main-panel.tsx

```diff
@@ -80,37 +75,121 @@ export const MainPanel = () => {
         focusElement(element);
     };
 
-    const handleDragEnd = (event: DragEndEvent) => {
-        const { active, over } = event;
+    const handleOnDragMove = useCallback(
+        debounce((event: DragEndEvent) => {
+            const wrapperBounds = wrapperRef.current?.getBoundingClientRect();
+            if (!wrapperBounds) return;
 
-        if (active.id !== over.id) {
-            const oldIndex = orderedIdentifiers.findIndex(
+            const wrapperLeftThirds =
+                wrapperBounds.left + wrapperBounds.width / 3;
+            const wrapperRightThirds =
+                wrapperBounds.left + (wrapperBounds.width * 2) / 3;
+            const mouseX =
+                event.active.rect.current.translated.left +
+                event.activatorEvent["layerX"];
+            const { active, over } = event;
+
+            const newIndex = orderedIdentifiers.findIndex(
                 (item) => item.internalId === active.id
             );
-            const newIndex = orderedIdentifiers.findIndex(
+            const oldIndex = orderedIdentifiers.findIndex(
                 (item) => item.internalId === over.id
             );
-            const updatedOrderedIdentifiers = arrayMove(
-                orderedIdentifiers,
-                oldIndex,
-                newIndex
-            );
 
+            let elementIdentifiers = [...orderedIdentifiers];
+            let updatedOrderedIdentifiers = [];
+
```

Comment:
thanks for the fix, I think the horizontal drag & drop generally works fine but it is becoming apparent that we need to handle the vertical drag & drop too. i.e. to determine whether admin is dragging to a new row or the current row, can we explore this?